File: WeatherStation-Java/WeatherStation.java
public class WeatherStation {
    public static void main(String[] args) {
        // 实例化 WeatherData
        WeatherData weatherData = new WeatherData();
        // 实例化 CurrentConditionsDisplay, ForecastDisplay
        CurrentConditionsDisplay currentDisplay = new CurrentConditionsDisplay(weatherData);
        ForecastDisplay forecastDisplay = new ForecastDisplay(weatherData);
        // 调用 setMeasurements 打印更新的数据
        weatherData.setMeasurements(26, 65, 30.4f);
        weatherData.setMeasurements(30, 50, 29.2f);
        weatherData.setMeasurements(18, 90, 29.2f);
    }
}

File: WeatherStation-Ruby/weather_station.rb
require_relative 'weather_data'
require_relative 'current_conditions_display'
require_relative 'forecast_display'
# 实例化 WeatherData
data = WeatherData.new
# 实例化 CurrentConditionsDisplay, ForecastDisplay
conditions_display = CurrentConditionsDisplay.new(data)
forecast_display = ForecastDisplay.new(data)
# 调用 set_measurements 打印更新的数据
data.set_measurements(26, 65, 30.4)
data.set_measurements(30, 50, 29.2)
data.set_measurements(18, 90, 30.6)
